File: frontend/src/api/graph.ts
import { api } from "./client";

export interface GraphNode {
  id: number;
  type: "milestone" | "issue";
  title: string;
  priority?: string;
  status?: string;
  issue_type?: string;
  labels?: string[];
  milestone_id?: number;
  parent_id?: number;
  size: number;
  color: string;
  opacity: number;
}

export interface GraphEdge {
  source: number;
  target: number;
}

export interface GraphResponse {
  nodes: GraphNode[];
  edges: GraphEdge[];
  labels: Record<string, string>;
}

export interface GraphParams {
  status?: string;
  issue_type?: string;
  labels?: string;
}

export const graphApi = {
  get: (slug: string, params?: GraphParams) =>
    api.get<GraphResponse>(`/projects/${slug}/graph`, { params }),
};

File: backend/src/schemas/graph.py
from typing import List, Optional, Dict
from pydantic import BaseModel


class GraphNode(BaseModel):
    id: str
    type: str
    title: str
    issue_id: Optional[int] = None
    priority: Optional[str] = None
    status: Optional[str] = None
    issue_type: Optional[str] = None
    labels: Optional[List[str]] = None
    milestone_id: Optional[int] = None
    parent_id: Optional[int] = None
    size: int = 12
    color: str = "#888888"
    opacity: float = 1.0


class GraphEdge(BaseModel):
    source: str
    target: str


class GraphResponse(BaseModel):
    nodes: List[GraphNode]
    edges: List[GraphEdge]
    labels: Dict[str, str]
